### PR TITLE
Add Stats Standalone App for Jetpack plugins

### PR DESCRIPTION
**Pull request by kangzj**
#### Proposed Changes
The Stats App is built to refresh the Stats experience in Jetpack. The counterpart of the project is in [here](https://github.com/Automattic/jetpack/pull/27247).

## Hierarchy
```
.
└── src/
    ├── components/       ← stats app only components. For now there is only a layout component.
    ├── page-middleware/  ← page.js integration with React and everything
    ├── app.js            ← entry point
    └── routes.js         ← page.js routes
```

### Production

```bash
cd apps/stats
yarn build
```
### Development

```bash
STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev
```

#### Testing Instructions
* Run `yarn start`
* Open `http://calypso.localhost:3000/stats/day/your-site`
* Ensure all stats pages work as expected

##### Optional Steps:
* Open your local Jetpack repository
* Checkout branch `add/stats-admin-package`
* Go back to Calypso repository
* Run `cd apps/stats`
* Run `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev` <- replace path
* Open `https://your-jetpack-site.com/wp-admin/admin.php?page=stats&calypso_stats=1`
* Ensure **most** pages work

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #

### CODE REVIEW

File: client/layout/masterbar/style.scss [hunk 1]
```diff
@@ -6,7 +6,7 @@ $masterbar-color-secondary: #101517;
 
 @import "@wordpress/base-styles/breakpoints";
 @import "@wordpress/base-styles/mixins";
-@import "client/assets/stylesheets/shared/animation";
+@import "../../assets/stylesheets/shared/animation";
```
Comment (jsnajdr) [inline client/layout/masterbar/style.scss]: Here `calypso/assets/...` is the correct path, consistent how we import things everywhere else. The `client/` prefix is a legacy one and this import was probably missed during the migration.

File: yarn.lock [hunk 2]
```diff
@@ -318,6 +318,38 @@ __metadata:
...
+    "@automattic/calypso-build": "workspace:^"
+    "@automattic/calypso-polyfills": "workspace:^"
+    "@automattic/webpack-extensive-lodash-replacement-plugin": "workspace:^"
+    "@automattic/webpack-inline-constant-exports-plugin": "workspace:^"
+    autoprefixer: ^10.2.5
+    calypso: "workspace:^"
+    enzyme: ^3.11.0
+    html-webpack-plugin: ^5.0.0-beta.4
+    jest: ^27.2.4
+    page: ^1.11.6
```
Comment (jsnajdr) [inline yarn.lock:336]: Main Calypso is still on `page@1.11.5`, let's be consistent with that, or upgrade `page` for the entire project. Then we won't get a duplicate `page@npm:^1.11.6` record in `yarn.lock`.

File: apps/stats/src/app.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import sites from 'calypso/state/sites/reducer';
+// import stats from 'calypso/state/stats/reducer';
+import { combineReducers, addReducerEnhancer } from 'calypso/state/utils';
+import registerStatsPages from './routes';
+import 'calypso/assets/stylesheets/style.scss';
+
+async function AppBoot() {
+	const rootReducer = combineReducers( {
+		currentUser,
+		sites,
```
Comment (jsnajdr) [inline apps/stats/src/app.jsx:32]: You'll probably need to add the `calypso/state/stats` reducer to make the Stats UIs really work. Without the Redux state, they won't be able to fetch stats data.

**Comment by kangzj** [comment]
@jsnajdr Thanks for the feedback ❤️ I'm still getting my head around this and will look at the points you made. Thanks 😄

File: yarn.lock (hunk 2 above)
Comment (kangzj) [reply yarn.lock:336]: Resolved in [f1e241a](https://github.com/Automattic/wp-calypso/pull/69231/commits/f1e241aaed02f7c110cc5496b369ed2e06657197)

File: client/layout/masterbar/style.scss (hunk 1 above)
Comment (kangzj) [reply client/layout/masterbar/style.scss]: Addressed in [0d4ace4](https://github.com/Automattic/wp-calypso/pull/69231/commits/0d4ace4c7a335ea2dd44444db80ad001212ef659)

File: apps/stats/src/app.jsx (hunk 3 above)
Comment (kangzj) [reply apps/stats/src/app.jsx:32]: stats states are automatically registered when use individual states by the stats components. Will look at see what we could do to make it explicit.

Comment (jsnajdr) [reply apps/stats/src/app.jsx:32]: Of course, most reducers are dynamically registered 🤦‍♂️ I forgot about it despite working on the dynamic registration myself 🙂

**Review by jsnmoon — APPROVED** [review]
Changes look reasonable and the existing Calypso Stats route appear to work as expected. Nice work!

**Pull request merged**